### PR TITLE
Stripe popup revamp

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Run: php artisan migrate

### CODE REVIEW

File: app/Http/Controllers/Common/PhpMailController.php [hunk 1]
```diff
@@ -253,14 +253,16 @@ public function setMailConfig($settings)
         }
     }
 
-    public function payment_log($from, $method, $status, $order, $exception = null)
+    public function payment_log($from, $method, $status, $order, $amount, $payment_type, $exception = null)
```
Comment (RafficMohammed) [inline app/Http/Controllers/Common/PhpMailController.php:256]: always add parameter at last. Don't add in between

File: database/migrations/2023_09_21_073606_add_order_id_to_auto_renewals_table.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        if (! Schema::hasColumn('order_id', 'payment_method')) {
+            Schema::table('auto_renewals', function (Blueprint $table) {
+                $table->string('order_id')->nullable();
```
Comment (RafficMohammed) [inline database/migrations/2023_09_21_073606_add_order_id_to_auto_renewals_table.php:18]: Order_id is string why ??. It's not foreign key

**Review by RafficMohammed — CHANGES_REQUESTED** [review]

**Review by sandesh556 — APPROVED** [review]

File: resources/views/themes/default1/layouts/front/master.blade.php [hunk 3]
```diff
@@ -467,12 +467,23 @@
...
 
             <div class="container">
+                
 
                   {!!Session::get('success')!!}
             </div>
 
             @endif
+            
+               @if(Session::has('Success'))
```
Comment (sandesh556) [inline resources/views/themes/default1/layouts/front/master.blade.php:477]: why was this added? we already have something to display success messages also a gurmeen PR is raised

**Review by sandesh556 — CHANGES_REQUESTED** [review]

File: resources/views/themes/default1/layouts/front/master.blade.php (hunk 3 above)
Comment (sowmiyasubramaniyan) [reply resources/views/themes/default1/layouts/front/master.blade.php:477]: For this @Gurmeenkour99 raised the PR, once that is merged I will rebase it.

**Review by sandesh556 — DISMISSED** [review]

**Review by RafficMohammed — APPROVED** [review]

**Review by SarfrazS — CHANGES_REQUESTED** [review]
- [ ] Issue1:- Remove increase decrease icon from card no filed.
- [ ] Issue2:- When we entered wrong card no with 16 digit, here shows correct but when we click on paynow getting below error.
image

- [ ] Issue3:- Put validation on month filed should not allow to enter more than 12.
- [ ] Issue4:- Stipe popup when we click out side popup not vanishing on auto-nenewal card details update page.

**Comment by sowmiyasubramaniyan** [comment]
> * [ ]  Issue1:- Remove increase decrease icon from card no filed.
> * [ ]  Issue2:- When we entered wrong card no with 16 digit, here shows correct but when we click on paynow getting below error.
>   image
> * [ ]  Issue3:- Put validation on month filed should not allow to enter more than 12.
> * [ ]  Issue4:- Stipe popup when we click out side popup not vanishing on auto-nenewal card details update page.

@SarfrazS Mentioned issues are fixed.

**Review by SarfrazS — CHANGES_REQUESTED** [review]
All Pointed issues were fixed but one new issue came Payment receipts are not showing.
image

File: app/Http/Controllers/Common/CronController.php [hunk 4]
```diff
@@ -854,20 +852,22 @@ private function getProcessingFee($paymentMethod, $currency)
...
 
         $mail = new \App\Http\Controllers\Common\PhpMailController();
         $mail->SendEmail($setting->email, $setting->company_email, $paymentSuccessdata, 'Payment Successful ');
-        $mail->payment_log($user->email, $payment, 'success', $order->number);
+        $mail->payment_log($user->email, $payment, 'success', $order->number, null, $amount, 'Product renew');
     }
 
     public function FailedPaymenttoAdmin($invoice, $total, $productName, $exceptionMessage, $user, $template, $order, $payment)
     {
+        $amount = currencyFormat($total, $user->currency);
```
Comment (sandesh556) [inline app/Http/Controllers/Common/CronController.php:866]: use country

File: app/Plugins/Stripe/Controllers/SettingsController.php [hunk 5]
```diff
@@ -162,6 +155,10 @@ public function postPaymentWithStripe(Request $request)
 
                 $control = new \App\Http\Controllers\Order\RenewController();
                 $cloud = new \App\Http\Controllers\Tenancy\CloudExtraActivities(new Client, new FaveoCloud());
+                if ($invoice->is_renewed == 0) {
+                    \Session::forget('subscription_id');
+                    \Session::forget('plan_id');
```
Comment (sandesh556) [inline app/Plugins/Stripe/Controllers/SettingsController.php:160]: Don't do this

**Review by sandesh556 — CHANGES_REQUESTED** [review]

**Review by sandesh556 — DISMISSED** [review]

**Review by SarfrazS — APPROVED** [review]
All Pointed issues are fixed.
First Round Testing Summary:-
Time taking in Hours/Days:- 8 Hours
Total No Of Issues:- 6
Minor: 6
Major: 0
Critical: 0
image
image
